### ISSUE TITLE
timeout-or-duplicatehostname-mismatch

### ISSUE THREAD

**Issue by szunyi** [raspgot/Contact-Form-PHP#10]
Hello,

As i see the closed issues this error is related to reCAPTCHA, but I dont use subdomains for my contact form.
I think I got this error when not touching the site and the reCAPTCHA is expired, but my biggest problem is in this error the contact form goes empty, all entered texts destory.

How can we deal this?

Thanks for your help, your project is really nice!

**Comment by raspgot** [comment]
Hello,

In case of domain alias you must register https://www.exemple.com AND https://exemple.com in the reCaptcha admin console

The code behind the reset of the form is in AjaxForm.js:70 ```javascript $("form").trigger("reset"); ``` but I recommend you to fill correctly your domain in the reCaptcha admin console

**Comment by szunyi** [comment]
its already added the www and the non-www version of the domain to the reCaptcha admin console

im using the cloudflare for this domain - maybe the can be cause this error while checking the host in the php side?

**Comment by raspgot** [comment]
Maybe
You can try to var_dump $_SERVER variables ?
```php
$resp = $recaptcha
    ->setExpectedHostname($_SERVER['SERVER_NAME'])
    ->verify($token, $_SERVER['REMOTE_ADDR']);
```

**Comment by raspgot** [comment]
Fix: 64ca9eab8124b047f3dde5e22b0dfb6ab69cce58
